MyCryptoHQ/MyCrypto
# Shapeshift Token Box UI

Closes #1277

### Description

Reverts the shapeshift token selector to a popover box. A few QoL changes along with that.

### Changes

* Change SwapDropdown to be a totally custom component, instead of using `<Select/>`
* Highlight "main" tokens above other tokens (ETH and BTC)
* Show all options and disable the current from currency, instead of hiding it
* Fix rates logic to re-enable TRST (Logic assumed 3 letter token names.)

### Steps to Test

1. Visit the following page: https://mycryptobuilds.com/5cdd8a205a6770d9fad9e3409d8b5586dcb032e2/#/swap
2. Open up from currency dropdown
3. Select ETH
4. Open up to currency dropdown
5. Confirm ETH is greyed out
6. Select preferred currency
7. Enter values and confirm swap submits
    * Known issue, inputs may look errorous but not actually be. This is on develop as well.
8. Confirm that token selector is still accessible with keyboard
9. Confirm that token selector looks good at mobile

## Screenshots

### From Token

![screen shot 2018-03-22 at 3 47 31 pm](https://user-images.githubusercontent.com/649992/37794845-233cbeea-2de9-11e8-98ba-1093f58bc135.png)

### To Token

![screen shot 2018-03-22 at 3 47 41 pm](https://user-images.githubusercontent.com/649992/37794848-2557e83a-2de9-11e8-8e36-d3d9d043141b.png)

### Mobile

![screen shot 2018-03-22 at 3 55 05 pm](https://user-images.githubusercontent.com/649992/37794944-6c8647ce-2de9-11e8-9aaf-3477d73a8f44.png)
